### PR TITLE
Tyler gps ping

### PR DESCRIPTION
**Pull request by redercj**
Tyler's changes for gps checking based on altitude.

---
This change is [Reviewable](https://reviewable.io/reviews/redercj/lightaprs-1.0/7)

### CODE REVIEW

File: LightAPRS-hab/LightAPRS-hab.ino [hunk 1]
```diff
@@ -183,9 +187,14 @@ void loop() {
   float tempC;
   float pressure;
   float loop_start = millis();
+  loopNumber++;
```
Comment (redercj) [inline LightAPRS-hab/LightAPRS-hab.ino:190]: should we consider putting this into any status or tx message?

File: LightAPRS-hab/LightAPRS-hab.ino [hunk 2]
```diff
@@ -255,14 +271,27 @@ void loop() {
...
     secsToCheckBatt--;
 
     secsToCheckBatt -= (millis()-loop_start)/1000;
     // sleepSeconds(BattWait-((millis-loop_start)/1000));
   }
-  sleepSeconds(secsTillTx);
-  secsTillTx = 0;
+  // Serial.println("Loop time in milliseconds->");
+  // Serial.println(round((millis()-loop_start)/1000));
+  int sleepSecs;
```
Comment (redercj) [inline LightAPRS-hab/LightAPRS-hab.ino:286]: let's move this to the start of the function and not define it in the middle.

File: LightAPRS-hab/LightAPRS-hab.ino [hunk 3]
```diff
@@ -255,14 +271,27 @@ void loop() {
...
+  if (secsTillPing <= secsTillTx) {
+    sleepSecs = secsTillPing;
+  } else {
+    sleepSecs = secsTillTx;
+  }
+  sleepSeconds(sleepSecs);
+  secsTillTx -= secsTillPing;
```
Comment (redercj) [inline LightAPRS-hab/LightAPRS-hab.ino:293]: let's say that secsTillPing is > secsTillTx.  What will happen here?

File: LightAPRS-hab/LightAPRS-hab.ino [hunk 4]
```diff
@@ -255,14 +271,27 @@ void loop() {
...
+  secsTillPing = 0;
```
Comment (redercj) [inline LightAPRS-hab/LightAPRS-hab.ino:294]: is this correct for the case where secsTillPing > secsTillTx ?

File: LightAPRS-hab/LightAPRS-hab.ino [hunk 5]
```diff
@@ -503,6 +532,7 @@ void updateTelemetry() {
   sprintf(telemetry_buff + 50, "%02d", gps.satellites.isValid() ? (int)gps.satellites.value() : 0);
   telemetry_buff[52] = 'S';
   telemetry_buff[53] = ' ';
+  telemetry_buff[54] =
```
Comment (redercj) [inline LightAPRS-hab/LightAPRS-hab.ino]: this got added inadvertently again.  could you please remove?

Comment (quantumbagel) [reply LightAPRS-hab/LightAPRS-hab.ino]: :) that was in my branch when tyler pulled

File: LightAPRS-hab/LightAPRS-hab.ino (hunk 2 above)
Comment (Pixelz22) [reply LightAPRS-hab/LightAPRS-hab.ino:286]: Why would we need to move it to the start? It doesn't get used till the very end, so why define it up at the beginning? Wouldn't that make the code more confusing?

Comment (redercj) [reply LightAPRS-hab/LightAPRS-hab.ino:286]: in C, many compilers would not even allow an inline declaration.  it is typically at the start of a brace segment, or beginning of the function. 

if (x==5) {
   int myVar;
   <blah>
}

but doing:

<blah>
int myVar;

<blah>

is generally not permitted/suggested/recommended and is, stylewise, not the best.

**Pull request merged**